app/uploaders/alert_image_uploader.rb

```diff
@@ -0,0 +1,57 @@
+class AlertImageUploader < ApplicationUploader
+  include CarrierWave::MiniMagick
+  include ::CarrierWave::Backgrounder::Delay
+
+  def store_dir
+    "#{base_store_dir}/#{model.id}"
+  end
+
+  def base_store_dir
+    "uploads/#{model.class.to_s[0, 2]}"
+  end
+
+  def filename
+    "alert_#{model.id}.jpg"
+  end
+
+  def extension_white_list
+    %w(jpg jpeg gif png tiff tif)
+  end
+
+  process :generate_alert_image
+  process convert: "jpg"
+
+  delegate :bike, to: :model
+
+  def bike_image
+    public_image = bike.public_images.first&.image
+    raise ArgumentError, "No bike image for Bike #{bike.id}" if public_image.blank?
+    public_image
+  end
+
+  def bike_url
+    "bikeindex.org/bikes/#{bike.id}"
+  end
+
+  def bike_location
+    city = bike.registration_address[:city]&.titleize
```

Comment:
I think this should grab the current stolen record theft location, and fallback to the registration address.

Reply:
Updated to prefer the stolen record's location:

```rb
# app/uploaders/alert_image_uploader.rb L33-39 (a0dc6a32)

def bike_location
  if stolen_record.location.present?
    stolen_record.location
  else
    bike.registration_location
  end
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/a0dc6a32/app/uploaders/alert_image_uploader.rb#L33-L39)]</sup>